tests/python_tests/test_llm_pipeline_npu.py

```diff
@@ -287,3 +289,150 @@ def test_chat_generation(config, model_id):
+
+    pipe = LLMPipeline(model_path, "NPU", **pipeline_config)
+    pipe.start_chat()
+    pipe.generate(tokenized_input_1, max_new_tokens=64, ignore_eos=True)
+
+    exception_pattern = r'Stateful LLM pipeline on NPU may only process prompts.*up to 64 ' +\
+                        r'tokens\..*[\n]?Set the "MAX_PROMPT_LEN" config option to increase ' +\
+                        r'the limit\.\n'
+    with pytest.raises(RuntimeError, match=exception_pattern):
+        pipe.generate(tokenized_input_2, max_new_tokens=2, ignore_eos=True)
+    # pipe.finish_chat()
+
+
+# FIXME: Should this work: call of generate() with tensor input in chat mode?
+@pytest.mark.skip(reason="Not verified that the use case is valid")
+@pytest.mark.precommit
+@pytest.mark.nightly
+@pytest.mark.parametrize("config", pipeline_configs)
+def test_terminate_by_too_long_history_for_tensor_inputs(config):
+    _, _, model_path = download_and_convert_model(get_models_list()[0])
+    pipeline_config = { "MAX_PROMPT_LEN": 64, "MIN_RESPONSE_LEN": 64 }
+    pipeline_config |= config
+
+    prompt_1 = 'The Sun is yellow because'
+    prompt_2 = 'Thank you for the explanation!'
+    tokenizer = Tokenizer(model_path)
+    tensor_input_1 = tokenizer.encode(prompt_1).input_ids
+    assert(isinstance(tensor_input_1, Tensor))
+    tensor_input_2 = tokenizer.encode(prompt_2).input_ids
+    assert(isinstance(tensor_input_2, Tensor))
+
+    pipe = LLMPipeline(model_path, "NPU", **pipeline_config)
+    pipe.start_chat()
+    pipe.generate(tensor_input_1, max_new_tokens=64, ignore_eos=True)
+
+    exception_pattern = r'Stateful LLM pipeline on NPU may only process prompts.*up to 64 ' +\
+                        r'tokens\..*[\n]?Set the "MAX_PROMPT_LEN" config option to increase ' +\
+                        r'the limit\.\n'
+    with pytest.raises(RuntimeError, match=exception_pattern):
+        pipe.generate(tensor_input_2, max_new_tokens=2, ignore_eos=True)
+    # pipe.finish_chat()
+
+
+@pytest.mark.precommit
+@pytest.mark.nightly
+@pytest.mark.parametrize("config", pipeline_configs)
+def test_terminate_by_too_long_history_for_string_inputs(config):
+    _, _, model_path = download_and_convert_model(get_models_list()[0])
+    pipeline_config = { "MAX_PROMPT_LEN": 64, "MIN_RESPONSE_LEN": 64 }
+    pipeline_config |= config
+
+    pipe = LLMPipeline(model_path, "NPU", **pipeline_config)
+
+    prompt_1 = 'The Sun is yellow because'
+    prompt_2 = 'Thank you for the explanation!'
+
+    pipe.start_chat()
+    pipe.generate(prompt_1, max_new_tokens=64, ignore_eos=True)
+
+    exception_pattern = r'Stateful LLM pipeline on NPU may only process prompts.*up to 64 ' +\
+                        r'tokens\..*[\n]?Set the "MAX_PROMPT_LEN" config option to increase ' +\
+                        r'the limit\.\n'
+    with pytest.raises(RuntimeError, match=exception_pattern):
+        pipe.generate(prompt_2, max_new_tokens=2, ignore_eos=True)
+    # pipe.finish_chat()
```

Comment:
It's up to you to call `finish_chat()` because `pipe` is destroyed after. But don't introduce commented code